### ISSUE TITLE
Path smoothing alternatives

### ISSUE THREAD

**Issue by blindape2** [microsoft/LightGBM#5144]
Just read #5139

I don't understand what is behind the formula to compute path_smooth option:
w_smoothed = (n / path_smooth) * w + w_p / (n / path_smooth + 1)
Being w the optimal weight for a leaf and w_p the weight of the parent (after applied path smoothing).

This formula can return much higher or much lower weights depending on the n (the number of observations in leaf) and path_smooth parameter. How is it affected by learning rate? Big size nodes will have in practice bigger learning rate equivalence.

**A desirable property:**

If w and w_p are very similar, the result v_smoothed should be like w and w_p. If w = w_p (that in practice never will happen because in that case we don't split), w_smoothed = w = w_p.
In general w_smoothed should be in the interval of both. There isn't any reason for having a w_smoothed out of this range. We can think of a weighted average of both values.
With the current formula we can get values outside the range.

There are several schemas for use that satisfy this property.

One intuitive way is **additive smoothing**: we compute w_smoothed adding to the leaf an amount of K average nodes of the parent node.
w_smoothed = (n * w + K * w_p) / (n + K)
In this case the amount of regularization depends of the size of leaf (n) and path_smooth parameter (K)

We may be interested in lower regularization when n is big. In that case we can use a second parameter p (p >= 1) to control the proportion of leaf and parent weights.
w_smoothed = (n^p * w + K * w_p) / (n^p + K)

**Comment by blindape2** [comment]
Just read #5139 and formula makes sense now.
